apm-sniffer/apm-sdk-plugin/shenyu-2.4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/shenyu/v24x/define/ServerWebExchangeInstrumentation.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.shenyu.v24x.define;
+
+import static net.bytebuddy.matcher.ElementMatchers.any;
+import static org.apache.skywalking.apm.agent.core.plugin.match.NameMatch.byName;
+
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.ClassInstanceMethodsEnhancePluginDefine;
+import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+/**
+ * ServerWebExchange Instrumentation.
+ */
+public class ServerWebExchangeInstrumentation extends ClassInstanceMethodsEnhancePluginDefine {
+
+    private static final String ENHANCE_CLASS = "org.springframework.web.server.adapter.DefaultServerWebExchange";
```

Comment:
This isn't a good practice. You are instrumenting `org.spingframework` in `shenyu` plugin, it could break others' use cases. Could you explain more?

Reply:
If I implement a whitelist method（witnessClasses()）, can I avoid interfering with other plugins?

Reply:
Yes, you can. But still, you should consider why this is a special case for Shenyu, rather than a general mechanism for spring ecosystem.(I don't know the detail, just asking)

Reply:
I've considered relying on apm-spring-weblux-5 X-plugin, but APM SDK plugin cannot directly rely on optional spring plugins.

Reply:
What do you mean `can't`? You just need to add a document to say, please activate this optional plugin. Nothing more.

Reply:
Spring gateway is working in this way, you could find it in the document.

Reply:
I got. This is a good practice. Let me reconstruct it

Reply:
It has been modified according to this good practice, and the local integration test has passed